File: tools/dbench/answers/go_haiku/s3/e2_missing_file.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

func main() {
	var cfg1 map[string]interface{}
	buf1, err1 := os.ReadFile("missing.cfg")
	if err1 != nil {
		fmt.Println("default")
	} else {
		json.Unmarshal(buf1, &cfg1)
		if val, exists := cfg1["name"]; exists {
			fmt.Println(val)
		} else {
			fmt.Println("default")
		}
	}

	var cfg2 map[string]interface{}
	portVal := 0
	buf2, err2 := os.ReadFile("app.cfg")
	if err2 == nil {
		json.Unmarshal(buf2, &cfg2)
		if p, exists := cfg2["port"]; exists {
			if pf, ok := p.(float64); ok {
				portVal = int(pf)
			}
		}
	}
	fmt.Println(portVal)
}

File: tools/dbench/tasks/e2_missing_file.py
import json
try:
    with open("missing.cfg") as f:
        cfg = json.load(f)
    name = cfg.get("name", "default")
except OSError:
    name = "default"
print(name)
with open("app.cfg") as f:
    cfg = json.load(f)
print(cfg.get("port", 0))
